src/main/java/cr0s/warpdrive/compat/CompatEnergyControl.java

```diff
@@ -0,0 +1,246 @@
+	private static final byte[] nbtRotFacing        = {     0,  1,  5,  4,  2,  3,  0,  1,  2,  3,  4,  5,  12, 13, 14, 15};
+	private static final byte[] horizontalRotFacing = {     1,  2,  3,  0,  5,  6,  7,  4,  8,  9,  10, 11, 12, 13, 14, 15};
+	
+	@Override
+	public int rotate(Block block, int metadata, NBTTagCompound nbtTileEntity, ITransformation transformation) {
+		final byte rotationSteps = transformation.getRotationSteps();
+		
+		
+		//Fix screen data for display panels
+		if(
+			classInfoPanel.isInstance(block) ||
+			classHoloPanel.isInstance(block)
+		){
+			NBTTagCompound screenData = nbtTileEntity.getCompoundTag("screenData");
+			int maxX = screenData.getInteger("maxX");
+			int maxY = screenData.getInteger("maxY");
+			int maxZ = screenData.getInteger("maxZ");
+			int minX = screenData.getInteger("minX");
+			int minY = screenData.getInteger("minY");
+			int minZ = screenData.getInteger("minZ");
+			BlockPos newMax = transformation.apply(maxX, maxY, maxZ);
+			BlockPos newMin = transformation.apply(minX, minY, minZ);
+			int newMaxX = Math.max(newMax.getX(), newMin.getX());
+			int newMaxY = Math.max(newMax.getY(), newMin.getY());
+			int newMaxZ = Math.max(newMax.getZ(), newMin.getZ());
+			int newMinX = Math.min(newMax.getX(), newMin.getX());
+			int newMinY = Math.min(newMax.getY(), newMin.getY());
+			int newMinZ = Math.min(newMax.getZ(), newMin.getZ());
+			
+			screenData.setInteger("maxX", newMaxX);
+			screenData.setInteger("maxY", newMaxY);
+			screenData.setInteger("maxZ", newMaxZ);
+			screenData.setInteger("minX", newMinX);
+			screenData.setInteger("minY", newMinY);
+			screenData.setInteger("minZ", newMinZ);
+			nbtTileEntity.setTag("screenData", screenData);
+		}
+
+		
+		//handle data cards (transform absolute coordinates)
+		if(
+			classHoloPanel.isInstance(block) ||
+			classInfoPanel.isInstance(block) ||
+			classRemoteThermalMonitor.isInstance(block) ||
+			classRangeTrigger.isInstance(block)
+		){
+			NBTTagList items = nbtTileEntity.getTagList("Items", 10);
+			NBTTagList itemsNew = items.copy();
+			boolean anyChange = false;
+			for(int index = 0; index < items.tagCount(); index++){
+				NBTTagCompound item = items.getCompoundTagAt(index);
+				if(!item.hasKey("tag")){ continue; }  //must have tag
+				NBTTagCompound itemTag = item.getCompoundTag("tag");
+				if(!(itemTag.hasKey("x") &&
+				     itemTag.hasKey("y") &&
+				     itemTag.hasKey("z"))){
+					continue;
+				}   //must have the coordinate data
+				
+				int x = itemTag.getInteger("x");
+				int y = itemTag.getInteger("y");
+				int z = itemTag.getInteger("z");
+				
+				if(!transformation.isInside(x, y, z)){ continue; }//only convert if inside the ship
+				BlockPos result = transformation.apply(x, y, z);
+				itemTag.setInteger("x", result.getX());
+				itemTag.setInteger("y", result.getY());
+				itemTag.setInteger("z", result.getZ());
+				item.setTag("tag", itemTag);
```

Comment:
This modify the original item, then overwrite it with a copy, consider skipping the copy and just modify in place? and by extension, no need to call `itemsNew.set()` and using `anyChange` boolean either...